### ISSUE TITLE
Retroarch 1.7.7 crashes on exit and messes monitor orientation (Ubuntu 18.04)

### ISSUE THREAD

**Issue by ramzeto** [libretro/RetroArch#9308]
## Description

I have a a two monitor setup, the primary one is 3840x2160 and the secondary is 1080x1920 (in portrait). You can see my monitor setup [here](https://www.dropbox.com/s/gruoet7eef89al3/screens_before.png?dl=1). Retroarch launches fine, everyting works, but when I exit the program, it crashes and messes the orientation of the primary monitor, changing it to portrait mode while preserving the same 3840x2160 resolution, see [here](https://www.dropbox.com/s/hxdr6bt9jnkvvj7/screens_after.png?dl=1). 

This does not happen with version 1.7.5.

### Expected behavior

To not crash on exit and mess the monitor orientation.

### Actual behavior

Crahes on exit and Messes the monitor orientation.

### Steps to reproduce the bug

1. Open retroarch
2. Close retroarch

### Bisect Results

Started happening when updating to 1.7.7. It happens with the version from the ppa (ppa:libretro/stable) and with [this version](https://github.com/libretro/RetroArch/releases/tag/v1.7.7) compiled from source by myself.

### Version/Commit

This is the retroarch --verbose output

[INFO] === Build =======================================
[INFO] CPU Model Name: Intel(R) Core(TM) i7-4790K CPU @ 4.00GHz
[INFO] Capabilities: MMX MMXEXT SSE1 SSE2 SSE3 SSSE3 SSE4 SSE4.2 AVX AES 
[INFO] Built: Aug 15 2019
[INFO] Version: 1.7.7
[INFO] =================================================
[INFO] Environ SET_PIXEL_FORMAT: RGB565.
[INFO] Redirecting save file to "/home/ram/Games/Retro Games/System/Retroarch/save_file/.srm".
[INFO] Redirecting savestate to "/home/ram/Games/Retro Games/System/Retroarch/save_state/.state".
[INFO] Version of libretro API: 1
[INFO] Compiled against API: 1
[INFO] [Audio]: Set audio input rate to: 30000.00 Hz.
[INFO] [Video]: Video @ fullscreen
[ERROR] [Wayland]: Failed to connect to Wayland server.
[INFO] [GLX]: GLX_OML_sync_control and GLX_MESA_swap_control supported, using better swap control method...
[INFO] [GL]: Found GL context: x
[INFO] [GL]: Detecting screen resolution 4920x2160.
[INFO] [GLX]: Window manager is GNOME Shell.
[INFO] [XINERAMA]: Xinerama version: 1.1.
[INFO] [XINERAMA]: Xinerama screens: 2.
[INFO] [GLX]: Using Xinerama on screen #0.
[INFO] [GLX]: X = 0, Y = 0, W = 3840, H = 2160.
[INFO] [GLX]: Using windowed fullscreen.
[INFO] [GLX]: Found swap function: glXSwapIntervalEXT.
[INFO] [GLX]: glXSwapIntervalEXT(1)
[INFO] [GL]: Vendor: X.Org, Renderer: Radeon RX 580 Series (POLARIS10, DRM 3.30.0, 5.1.0-050100-generic, LLVM 8.0.0).
[INFO] [GL]: Version: 4.5 (Compatibility Profile) Mesa 19.0.8.
[INFO] [GL]: Using ARB_sync to reduce latency.
[INFO] [GL]: Using resolution 3840x2160
[INFO] [GL]: Default shader backend found: glsl.
[INFO] [Shader driver]: Using GLSL shader backend.
[INFO] [GLSL]: Checking GLSL shader support ...
[WARN] [GL]: Stock GLSL shaders will be used.
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Linking GLSL program.
[INFO] Setting up menu pipeline shaders for XMB ... 
[INFO] [GLSL]: Compiling ribbon shader..
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GLSL]: Compiling simple ribbon shader..
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GLSL]: Compiling snow shader..
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GLSL]: Compiling modern snow shader..
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GLSL]: Compiling bokeh shader..
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GLSL]: Compiling snowflake shader..
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GL]: Using 4 textures.
[INFO] [GL]: Loaded 1 program(s).
[INFO] [GL]: Using GL_RGB565 for texture uploads.
[INFO] [Device]: Found pad: PDP Wired Controller for Xbox One - Camo Series on /dev/input/js0.
[INFO] [Joypad]: Found joypad driver: "linuxraw".
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [Video]: Found display server: x11
[INFO] [PulseAudio]: Requested 24576 bytes buffer, got 18432.
[INFO] [Menu]: Found menu display driver: "gl".
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [Menu]: Found menu display driver: "gl".
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [LED]: LED driver = 'null' 0x5567fa081ba0
[INFO] [MIDI]: Initializing ...
[INFO] [MIDI]: Input disabled.
[INFO] [MIDI]: Output disabled.
[INFO] [MIDI]: Initialized "null" driver.
[INFO] SRAM will not be saved.
[INFO] Loading history file: [/home/ram/.config/retroarch/content_history.lpl].
[INFO] Loading history file: [/home/ram/.config/retroarch/content_favorites.lpl].
[INFO] Loading history file: [/home/ram/.config/retroarch/content_music_history.lpl].
[INFO] Loading history file: [/home/ram/.config/retroarch/content_video_history.lpl].
[INFO] Loading history file: [/home/ram/.config/retroarch/content_image_history.lpl].
[INFO] [GL]: VSync => on
[INFO] [GLX]: glXSwapIntervalEXT(1)
[INFO] [GL]: VSync => on
[INFO] [GLX]: glXSwapIntervalEXT(1)
[INFO] [PulseAudio]: Pausing.
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [Font]: Using font rendering backend: freetype.
[INFO] [PulseAudio]: Unpausing.
[INFO] [config] Saved new config to "/home/ram/.config/retroarch/retroarch.cfg".
[INFO] Content ran for a total of: 00 hours, 00 minutes, 00 seconds.
[INFO] Unloading game..
[INFO] [PulseAudio]: Pausing.
[INFO] Unloading core..
[INFO] Unloading core symbols..
[INFO] [XINERAMA]: Xinerama version: 1.1.
[INFO] [XINERAMA]: Xinerama screens: 2.
[INFO] [XINERAMA]: Saved monitor #0.
[INFO] [Video]: Does not have enough samples for monitor refresh rate estimation. Requires to run for at least 4096 frames.
[INFO] [Video]: Does not have enough samples for monitor refresh rate estimation. Requires to run for at least 4096 frames.
[INFO] [Video]: Setting screen orientation to 3.
X Error of failed request:  BadMatch (invalid parameter attributes)
  Major opcode of failed request:  140 (RANDR)
  Minor opcode of failed request:  7 (RRSetScreenSize)
  Serial number of failed request:  19
  Current serial number in output stream:  20

### Environment information

I am running Ubuntu 18.04 with kernel 5.1. This is the output of _uname -a_:
Linux ram-desktop 5.1.0-050100-generic #201905052130 SMP Mon May 6 01:32:59 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

The Mesa library version is 19.0.8. The output of _glxinfo | grep "OpenGL version"_ is:
OpenGL version string: 4.5 (Compatibility Profile) Mesa 19.0.8

My GPU is an RX 580, I am using the amdgpu driver. This is the output of _lspci -k | grep -EA3 'VGA|3D|Display'_:
01:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Ellesmere [Radeon RX 470/480/570/570X/580/580X] (rev e7)
	Subsystem: XFX Pine Group Inc. Ellesmere [Radeon RX 470/480/570/580]
	Kernel driver in use: amdgpu
	Kernel modules: amdgpu

gcc and g++ versions are the ones shipped with Ubuntu. The _g++ --version_ output is:
g++ (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0

_gcc --version_ output is:
gcc (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0

For reference, this is the output of the _./configure_:

Checking operating system ... Linux (Ubuntu 18.04.3 LTS 18.04)
Checking for suitable working C compiler ... /usr/bin/gcc works
Checking for suitable working C++ compiler ... /usr/bin/g++ works
Checking for pkg-config ... /usr/bin/pkg-config
Checking for availability of switch -std=gnu99 in /usr/bin/gcc ... yes
Checking for availability of switch -Wno-unused-result in /usr/bin/gcc ... yes
Checking for availability of switch -Wno-unused-variable in /usr/bin/gcc ... yes
Checking function sd_get_machine_names in -lsystemd ... no
Checking presence of package bcm_host ... no
Checking function bcm_host_init in -lbcm_host ... no
Checking presence of package egl ... 19.0.8
Checking function ass_library_init in -lass ... no
Checking function pthread_create in -lpthread ... yes
Checking function pthread_key_create in -lpthread ... yes
Checking function dlopen in -ldl ... yes
Checking function socket in -lc ... yes
Checking function getaddrinfo in -lc ... yes
Checking existence of -lminiupnpc ... no
Checking function fcntl in -lc ... yes
Checking function getopt_long in -lc ... yes
Checking presence of package alsa ... 1.1.3
Checking presence of predefined macro AUDIO_SETINFO in sys/audioio.h ... no
Checking presence of header file sys/soundcard.h ... yes
Checking presence of header file soundcard.h ... no
Checking existence of -lossaudio ... no
Checking function alcOpenDevice in -lopenal ... no
Checking presence of package rsound >= 1.1 ... no
Checking presence of package libroar >= 1.0.12 ... no
Checking presence of package jack >= 0.120.1 ... no
Checking presence of package libpulse ... 11.1
Checking presence of package sdl >= 1.2.10 ... no
Checking presence of package sdl2 >= 2.0.0 ... 2.0.8
Checking presence of package Qt5Core >= 5.2 ... 5.9.5
Checking presence of package Qt5Gui >= 5.2 ... 5.9.5
Checking presence of package Qt5Widgets >= 5.2 ... 5.9.5
Checking presence of package Qt5Concurrent >= 5.2 ... 5.9.5
Checking presence of package Qt5Network >= 5.2 ... 5.9.5
Checking presence of package openssl >= 1.0.0 ... 1.0.2n
Checking presence of package flac ... no
Checking presence of header file mbedtls/entropy.h ... no
Checking existence of -lmbedx509 ... no
Checking existence of -lmbedcrypto ... no
Checking presence of package libusb-1.0 >= 1.0.13 ... 1.0.21
Checking presence of header file GL/gl.h ... yes
Checking existence of -lGL ... yes
Checking function cgCreateContext in -lCg -lCgGL ... yes
Checking presence of package zlib ... 1.2.11
Checking presence of package libavcodec >= 54 ... 57.107.100
Checking presence of package libavformat >= 54 ... 57.83.100
Checking presence of package libavdevice ... 57.10.100
Checking presence of package libswresample ... 2.9.100
Checking presence of package libavresample ... 3.7.0
Checking presence of package libavutil >= 51 ... 55.78.100
Checking presence of package libswscale >= 2.1 ... 4.8.100
Checking presence of header file libavutil/channel_layout.h ... yes
Checking function dlopen in -ldl ... yes
Checking presence of package gbm >= 9.0 ... no
Checking presence of package libdrm ... 2.4.97
Checking presence of package vg ... no
Checking presence of package libv4l2 ... no
Checking presence of package freetype2 ... 21.0.15
Checking presence of package x11 ... 1.6.4
Checking presence of package xcb ... 1.13
Checking presence of package wayland-egl >= 10.1.0 ... 18.1.0
Checking presence of package wayland-cursor >= 1.12 ... 1.16.0
Checking presence of package wayland-protocols >= 1.15 ... no
Checking presence of package wayland-scanner >= 1.12 ... 1.16.0
Checking presence of package xkbcommon >= 0.3.2 ... 0.8.0
Checking presence of package xext ... 1.3.3
Checking presence of package xxf86vm ... 1.1.4
Notice: Using the bundled wayland-protocols.
Checking existence of -lXrandr ... yes
Checking presence of package xinerama ... 1.1.3
Checking presence of package xv ... 1.0.11
Checking presence of package libudev ... 237
Checking presence of header file linux/parport.h ... yes
Checking presence of header file linux/ppdev.h ... yes
Checking function strcasestr in -lc ... yes
Checking function mmap in -lc ... yes
Checking function vkCreateInstance in -lvulkan ... yes
Checking for moc ... /usr/bin/moc works
Creating make config: config.mk
Creating config header: config.h

**Comment by RobLoach** [comment]
I haven't been able to reproduce this... But I'm on MATE. Are you toggling fullscreen off/on?

**Comment by LazyBumHorse** [comment]
Can't seem to reproduce this as well (on nightly), though I don't have a second monitor.

Maybe RA 1.7.5 didn't have the Screen Orientation setting, so it simply ignored the orientation(?)

> [INFO] [Video]: Setting screen orientation to 3.
> X Error of failed request: BadMatch (invalid parameter attributes)
> Major opcode of failed request: 140 (RANDR)
> Minor opcode of failed request: 7 (RRSetScreenSize)
> Serial number of failed request: 19
> Current serial number in output stream: 20

That's a bit odd, orientation 3 seems to be Portrait Right or 270° in RA, why would that be a BadMatch?
Or another question: Why does it try to set the screen orientation at all?
Shouldn't RA keep the system default?

@ramzeto What does Settings -> Video -> Screen Orientation say? Can you try if setting it to "Normal" and save the configuration under Configuration File -> Save Current Configuration before restarting RA? (Alternatively, set `screen_orientation = "0"` in your retroarch.cfg)

**Comment by LazyBumHorse** [comment]
Here, that's what I thought, when `screen_orientation = "0"`, it doesn't set the screen orientation on init
https://github.com/libretro/RetroArch/blob/d96d80cf5cb32e30318140a686981ce4b2176919/retroarch.c#L17171-L17172

And it shouldn't set it exit as well, but maybe it does. Relevant code:
https://github.com/libretro/RetroArch/blob/d96d80cf5cb32e30318140a686981ce4b2176919/gfx/video_display_server.c#L73-L76

**Comment by ramzeto** [comment]
Hi, thank you for your responses.

@RobLoach  Retroarch 1.7.7 creashed when toggling the fullscreen switch. It does not even save the setting. If I want to change this, I have to edit the retroarch.cfg. The crash also happens in window mode.

@LazyBumHorse my screen_orientation setting has always been 0.

For testing, I disabled my secondary monitor and the problem went away. Maybe there is something with a multi monitor setup that messes the initial setup of some variables.

With my secondary monitor disabled, this message does not appear anymore:

```
[INFO] [Video]: Setting screen orientation to 3.
X Error of failed request: BadMatch (invalid parameter attributes)
Major opcode of failed request: 140 (RANDR)
Minor opcode of failed request: 7 (RRSetScreenSize)
Serial number of failed request: 19
Current serial number in output stream: 20
```

**Comment by LazyBumHorse** [comment]
I looked at the code again and what we are seeing here is simply not possible.
So I looked for code changes - et voilá: https://github.com/libretro/RetroArch/commit/c39765a5b7e95926caa72423a9cf538ad7906be6

This one should fix the problem you're having because it will only try to set the orientation back if it was changed in the first place.
You simply need to update to nightly or wait for 1.7.8 which might release soon.

There is however still the open question why setting the orientation _does_ fail.
I'm pretty sure you can crash RetroArch by just going into the video settings and change the orientation, is that right?

**Comment by ramzeto** [comment]
@LazyBumHorse sorry for  the delay.

Yes, it crashes the moment I try to change the orientation.

I have done another test, this time I set both of my monitors in landscape orientation. With this setup Retroarch works flawlessly, the problem goes away and the **X Error of failed request: BadMatch (invalid parameter attributes) ...** message does not appear.

This makes me believe that it is considering the orientation of any of the displays connected to the computer rather than considering only the orientation of the display matching  the **video_monitor_index** setting.

**Comment by badookey** [comment]
I do not see this issue as closed, so would like to report I had the exact same issue with the same context of dual screens with different orientations, see below:

[INFO] [Video]: Setting screen orientation to 3.
X Error of failed request:  BadMatch (invalid parameter attributes)
  Major opcode of failed request:  140 (RANDR)
  Minor opcode of failed request:  7 (RRSetScreenSize)
  Serial number of failed request:  22
  Current serial number in output stream:  23

This was fixed by updating from 1.7.7 to 1.8.0-1.
